core/src/test/java/io/confluent/rest/SslFactoryTest.java

```diff
@@ -0,0 +1,112 @@
+package io.confluent.rest;
+
+import org.apache.kafka.common.config.SslConfigs;
+import org.apache.kafka.common.config.types.Password;
+import org.apache.kafka.common.errors.InvalidConfigurationException;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.junit.jupiter.api.Test;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class SslFactoryTest {
+  private static String CA1;
+  private static String CA2;
+  private static String CERTCHAIN;
+  private static String KEY;
+  private static String ENCRYPTED_KEY;
+  private static final String PEM_TYPE = "PEM";
+  private static final String PEM_KEYSTORE_NAME_PREFIX = "pemKeyStore";
+  private static final String PEM_KEYSTORE_NAME_SUFFIX = ".pem";
+
+  static {
+    try {
+      Path path = Paths.get(Objects.requireNonNull(
+          SslFactory.class.getClassLoader().getResource("certs/cert1.pem").toURI()));
```

Comment:
as already commented, please use code to generate certificates/private keypair whenever possible.

Reply:
@trnguyencflt 
The code to generate is quite complicated and will require bouncy castle libs.
Most recommendations are to use openssl locally and use the values produced.

Reply:
> The code to generate is quite complicated and will require bouncy castle libs.

Do you mean it requires non-fips libs to generate?

Is this the same method that we use in ce-kafka PR?

Reply:
> Most recommendations are to use openssl locally and use the values produced.

Could you please add the instruction to run openssl in a README file in resources/certs?

Reply:
@trnguyencflt 

* cert1.pem and cert2.pem expiry is 2120 (**97 yrs** should be enough to not need regeneration)

* Yes, current code in ce-kafka uses non-fips (BC) provider to generate artifacts (`CertificateUtils`). As part of aforementioned PR in ce-kafka I've added changed to support FIPS provider - but having that will bring in a new dependency (ce-fips-tests), which can be avoided this way.

* The instructions to create root pair (root CA and private key) and intermediate certs are described in:
https://openssl-ca.readthedocs.io/en/latest/create-the-root-pair.html
https://openssl-ca.readthedocs.io/en/latest/create-the-intermediate-pair.html